tests/rptest/services/redpanda_installer.py

```diff
@@ -639,36 +647,16 @@ def _install_unlocked(self, nodes, version):
         if self._nodes_share_installs:
             nodes_to_download = [nodes[0]]
 
-        ssh_download_per_node = dict()
-        for node in nodes_to_download:
-            if not version == RedpandaInstaller.HEAD and not node.account.exists(
-                    version_root):
-                ssh_download_per_node[
-                    node] = self._async_download_on_node_unlocked(
-                        node, version)
-
-        try:
-            self.wait_for_async_ssh(self._redpanda.logger,
-                                    ssh_download_per_node,
-                                    "Finished downloading binaries")
-        except Exception as e:
-            self._redpanda.logger.error(
-                f"Exception while downloading to {version_root}, cleaning up: {str(e)}"
-            )
-            # TODO: make failure handling more fine-grained. If deploying on
-            # dedicated nodes, we only need to clean up the node that failed.
-            for node in ssh_download_per_node:
-                ssh_iter = ssh_download_per_node[node]
-                if ssh_iter.has_next():
-                    # Drain the iterator to make sure we wait for on-going
-                    # downloads to finish before cleaning up.
-                    try:
-                        [l for l in ssh_iter]
-                    except:
-                        pass
-                # Be permissive so we can clean everything.
-                node.account.remove(version_root, allow_fail=True)
-            raise e
+        # Download in parallel.
+        with ThreadPoolExecutor() as executor:
+            pending_f = []
+            for node in nodes_to_download:
+                if not version == RedpandaInstaller.HEAD:
+                    pending_f.append(
+                        executor.submit(self.download_on_node_unlocked, node,
+                                        version))
+            for f in pending_f:
+                f.result()
```

Comment:
Curious if it's important we do anything if this throws?

Reply:
I don’t believe there is anything to do with cleanup is now handled inside the download_on_node_unlocked. Here we just propagate the error and let test fail.

But you made me think about something else. What if one throws but other still download? Does ducktape kill all the threads in the middle? Do I need to wait on all futures first and only after all are “resolved” propagate the error? 🤔

Reply:
>What if one throws but other still download?

Yea this was more what I was wondering. I'm not sure 🤔

Reply:
context manager (created by `with` statement) will wait for all futures to resolve https://trinket.io/python3/deb6edcfae81